### ISSUE TITLE
[website] datathinking.org/principles

### ISSUE THREAD

**Issue by altosaar** [onefact/datathinking.org#121]
- be clear
- be specific
- write it down

recognize it's hard to assess:
1. how you feel
2. what you want to do

there are many factors at play, and tools like chatgpt can help you receive rapid feedback on which stage of the process you are at.

for example, are you clarifying a research question? identifying an unmet need in a population of stakeholders? designing a user interface to meet a need using artificial intelligence?

**Comment by altosaar** [comment]
`OK, sorry! We changed our mind and we'd rather receive bounding boxes for the ends of a street in the jpg images. Can you please revise this?`

claude:

That is a quite different task! For street segmentation and finding end points, you'll want a semantic segmentation model rather than an object detector.
Here is an example using PyTorch and a pretrained segmentation model (DeepLabV3):
model = torch.hub.load('pytorch/vision:v0.10.0', 'deeplabv3_resnet101', pretrained=True) 
model.eval()

img = torch.rand(3, 800, 1200)  # sample input image
output = model(img)

# Get pixel predictions - each value is a class ID
predictions = output['out'].argmax(1)  

# Find endpoints where road class transitions to non-road
endpoints = []
for row in range(predictions.shape[1]):
    for col in range(predictions.shape[2] - 1):
        if predictions[0, row, col] != predictions[0, row, col + 1]:
            endpoints.append((row, col))  # (y, x) coordinate

print(endpoints)  # e.g. [(223, 562), (401, 1103), ...]

So the endpoints list will contain (y, x) coords for all the spots where the road ends.
Again, you'd feed an actual street image into img, and the rest would be largely the same. Let me know if you have any other questions!

**Comment by altosaar** [comment]
follow the money, follow the data, follow the status (kissinger, openai, etc)

understand incentives, behavioral economics, psychology, and ethnography to understand provenance in-context

 parsimoniony: 
- if it doesn't work for me, it won't work for them
- if it doesn't work here, it won't work there
focus on edge cases to ensure data thinking artefacts are simple

focus on low latency: balance compute with performance

validate with computers and humans 

math before code

be clear, be specific, write it down.

self-monitoring for behavior change.

feel, ask, think, do

start with the math

visualizations: visualize early, visualize often

style: be fancy, principled, aesthetic. axes; ismael's.

great data thinkers steal.

data: FAIR

tech: use simplest tools that are most composable and most open source and most accessible worldwide.

**Comment by altosaar** [comment]
if it's too complicated, make it simpler (mindshare). the market doesn't reward waiting.

**Comment by altosaar** [comment]
have one source of ground truth and understand its provenance, governance, and metadata. look at it often and reflect on what unknown unknowns you may be missing.

**Comment by altosaar** [comment]
Articulate things clearly.

Articulate status games, optics, and incentives that drive data and drive thinking clearly.

**Comment by altosaar** [comment]
Teach someone to see the impact of your actions and principles on others (prespcriptive vs descriptive)(

**Comment by altosaar** [comment]
focus on etymology in naming a la websters. / jsumers
